Modules/_lsprof.c

```diff
@@ -87,7 +88,14 @@ _lsprof_get_state(PyObject *module)
 
 static PyTime_t CallExternalTimer(ProfilerObject *pObj)
 {
-    PyObject *o = _PyObject_CallNoArgs(pObj->externalTimer);
+    PyObject *o = NULL;
+
+    // External timer can do arbitrary things so we need a flag to prevent
+    // horrible things to happen
+    pObj->flags |= POF_EXT_TIMER;
+    o = _PyObject_CallNoArgs(pObj->externalTimer);
+    pObj->flags &= ~POF_EXT_TIMER;
```

Comment:
Would not using a strong reference to external timer help?
```py
    PyObject *externalTimer = Py_NewRef(pObj->externalTimer);
    o = _PyObject_CallNoArgs(externalTimer);
    Py_DECREF(externalTimer);
```

Reply:
I don't quite understand the purpose here.

Reply:
I see, this does not help much. The crash happens when `ProfilerObject` is deleted.

Then I think that your solution is the best.